frontend/src/metabase/nav/components/PaymentBanner/PaymentBanner.unit.spec.tsx

```diff
@@ -0,0 +1,126 @@
+/* eslint jest/expect-expect: ["error", { "assertFunctionNames": ["expect", "expectBannerToBeHidden"] }] */
+import { render, screen } from "__support__/ui";
+
+import type { TokenStatusStatus } from "metabase-types/api";
+
+import { PaymentBanner, shouldRenderPaymentBanner } from "./PaymentBanner";
+
+interface SetupOpts {
+  isAdmin: boolean;
+  tokenStatusStatus: TokenStatusStatus;
+}
+
+function setup({ isAdmin, tokenStatusStatus }: SetupOpts) {
+  render(
+    <PaymentBanner isAdmin={isAdmin} tokenStatusStatus={tokenStatusStatus} />,
+  );
+}
+
+const PAST_DUE_BANNER_REGEX =
+  /We couldn't process payment for your account. Please/;
+
+const UNPAID_BANNER_REGEX =
+  /Pro features won’t work right now due to lack of payment\./;
+
+describe("PaymentBanner", () => {
+  describe("admin users", () => {
+    it("should render past-due banner when token status status is `past-due`", () => {
+      setup({
+        isAdmin: true,
+        tokenStatusStatus: "past-due",
+      });
+
+      expect(screen.getByText(PAST_DUE_BANNER_REGEX)).toBeInTheDocument();
+    });
+
+    it("should render unpaid banner when token status status is `unpaid`", () => {
+      setup({
+        isAdmin: true,
+        tokenStatusStatus: "unpaid",
+      });
+
+      expect(screen.getByText(UNPAID_BANNER_REGEX)).toBeInTheDocument();
+    });
+
+    it("should not render any banner when token status status is not `past-due` or `unpaid`", () => {
+      setup({
+        isAdmin: true,
+        tokenStatusStatus: "something-else",
+      });
+
+      expectBannerToBeHidden();
+    });
+  });
+
+  describe("non-admin users", () => {
+    it.each([
+      {
+        tokenStatusStatus: "past-due",
+      },
+      {
+        tokenStatusStatus: "unpaid",
+      },
+      {
+        tokenStatusStatus: "something-else",
+      },
+    ])(
+      "should not render any banner when tokenStatusStatus is `$tokenStatusStatus`",
+      ({ tokenStatusStatus }) => {
+        setup({ isAdmin: false, tokenStatusStatus });
+
+        expectBannerToBeHidden();
+      },
+    );
+  });
+
+  describe("shouldRenderPaymentBanner", () => {
+    it.each([
+      {
+        isAdmin: true,
+        tokenStatusStatus: "past-due",
+        shouldRenderPaymentBanner: true,
+      },
+      {
+        isAdmin: true,
+        tokenStatusStatus: "unpaid",
+        shouldRenderPaymentBanner: true,
+      },
+      {
+        isAdmin: true,
+        tokenStatusStatus: "something-else",
+        shouldRenderPaymentBanner: false,
+      },
+      {
+        isAdmin: false,
+        tokenStatusStatus: "past-due",
+        shouldRenderPaymentBanner: false,
+      },
+      {
+        isAdmin: false,
+        tokenStatusStatus: "unpaid",
+        shouldRenderPaymentBanner: false,
+      },
+      {
+        isAdmin: false,
+        tokenStatusStatus: "something-else",
+        shouldRenderPaymentBanner: false,
```

Comment:
I'd call it something different to avoid the collision, e.g. `hasBanner` or something like this

Reply:
Fixed